src/js/views/portals/PortalView.js

```diff
@@ -664,27 +666,33 @@ define([
        * Update the window location path with the active section name
        * @param {boolean} [showSectionLabel] - If true, the section label will be added to the path
        */
-      updatePath: function (showSectionLabel) {
-        var label = this.model.get("label") || this.newPortalTempName,
-          originalLabel =
-            this.model.get("originalLabel") || this.newPortalTempName,
-          pathName = decodeURIComponent(window.location.pathname)
-            .substring(MetacatUI.root.length)
-            // remove trailing forward slash if one exists in path
-            .replace(/\/$/, "");
+      updatePath(showSectionLabel) {
+        const label = this.model.get("label") || this.newPortalTempName;
+        const originalLabel =
+          this.model.get("originalLabel") || this.newPortalTempName;
+        const pathName = decodeURIComponent(window.location.pathname)
+          .substring(MetacatUI.root.length)
+          // remove trailing forward slash if one exists in path
+          .replace(/\/$/, "");
 
         // Add or replace the label and section part of the path with updated values.
         // pathRE matches "/label/section", where the "/section" part is optional
-        var pathRE = new RegExp(
+        const pathRE = new RegExp(
           "\\/(" + label + "|" + originalLabel + ")(\\/[^\\/]*)?$",
           "i",
         );
-        newPathName = pathName.replace(pathRE, "") + "/" + label;
+        let newPathName = pathName.replace(pathRE, "") + "/" + label;
 
         if (showSectionLabel && this.activeSection) {
           newPathName += "/" + this.activeSection.uniqueSectionLabel;
         }
 
+        const searchQueryString = new URL(window.location.href).search;
+        // Support optional parameters for loading a portal's view from URL.
+        if (searchQueryString) {
+          newPathName += searchQueryString;
+        }
```

Comment:
Hmm I'm not really sure what to do here. Unfortunately this function is called when the page loads, and without it, it deletes all of the query parameters.

Reply:
The experience is better now, however, since I set the query parameters only on initialization and on interaction, if the user goes to a different tab, then jumps back to the image viewer, there will be no query parameters since the view has already been initialized. The user would need to interact with the map in someway for the parameters to come back. Are you aware of any callback for when a view becomes "active" again? @robyngit @yvonnesjy

Reply:
Okay, I added a chain of function calls and an event trigger, it solve this problem fairly well, and supports future cases where we can take action on section change. Right now only the PortalVizSectionView actually does anything, but all PortalSectionViews are aware of the change